scripts/slack_parameter_setup.py

```diff
@@ -0,0 +1,403 @@
+
+        Args:
+            token: The bot token to validate
+
+        Returns:
+            Tuple of (is_valid, error_message)
+        """
+        if not token:
+            return False, "Bot token cannot be empty"
+
+        if not self.BOT_TOKEN_PATTERN.match(token):
+            return (
+                False,
+                "Bot token must be in format: xoxb-XXXXXXXXX-XXXXXXXXX-XXXXXXXXXXXXXXXX",
+            )
+
+        return True, ""
+
+    def validate_signing_secret(self, secret: str) -> Tuple[bool, str]:
+        """
+        Validate Slack Signing Secret format.
+
+        Args:
+            secret: The signing secret to validate
+
+        Returns:
+            Tuple of (is_valid, error_message)
+        """
+        if not secret:
+            return False, "Signing secret cannot be empty"
+
+        if not self.SIGNING_SECRET_PATTERN.match(secret):
+            return False, "Signing secret must be a 32-character hexadecimal string"
+
+        return True, ""
+
+    def validate_workspace_id(self, workspace_id: str) -> Tuple[bool, str]:
+        """
+        Validate Slack Workspace ID format.
+
+        Args:
+            workspace_id: The workspace ID to validate
+
+        Returns:
+            Tuple of (is_valid, error_message)
+        """
+        if not workspace_id:
+            return False, "Workspace ID cannot be empty"
+
+        if not self.WORKSPACE_ID_PATTERN.match(workspace_id):
+            return (
+                False,
+                "Workspace ID must be 9-11 uppercase alphanumeric characters",
+            )
+
+        return True, ""
+
+    def create_or_update_parameter(
+        self,
+        parameter_name: str,
+        parameter_value: str,
+        description: str,
+        overwrite: bool = False,
+    ) -> bool:
+        """
+        Create or update an SSM parameter with encryption.
+
+        Args:
+            parameter_name: The parameter name/path
+            parameter_value: The parameter value
+            description: Description of the parameter
+            overwrite: Whether to overwrite existing parameter
+
+        Returns:
+            True if successful, False otherwise
+        """
+        try:
+            self.ssm_client.put_parameter(
+                Name=parameter_name,
+                Value=parameter_value,
+                Description=description,
+                Type="SecureString",
+                Overwrite=overwrite,
+                Tags=[
+                    {"Key": "Integration", "Value": "SlackSecurityIR"},
+                    {"Key": "ManagedBy", "Value": "SlackParameterSetup"},
+                    {"Key": "LastRotated", "Value": datetime.utcnow().isoformat()},
+                ],
+            )
+            print("✅ Successfully created/updated parameter")
+            return True
+        except ClientError as e:
+            error_code = e.response["Error"]["Code"]
+            if error_code == "ParameterAlreadyExists" and not overwrite:
```

Comment:
Question: I'm confused. the method is `create_or_update` but we aren't handling a update? Why wouldn't we just call `rotate` if this exception gets triggered.